lib/screens/surveydetails/widget/answer_multi_choice_widget.dart

```diff
@@ -0,0 +1,118 @@
+import 'package:flutter/material.dart';
+import 'package:lydiaryanfluttersurvey/model/ui/question_ui_model.dart';
+import 'package:lydiaryanfluttersurvey/resources/dimensions.dart';
+
+class AnswerMultiChoiceWidget extends StatefulWidget {
+  final QuestionUiModel question;
+  final Function(List<String>) onChecked;
+
+  const AnswerMultiChoiceWidget({
+    Key? key,
+    required this.question,
+    required this.onChecked,
+  }) : super(key: key);
+
+  @override
+  State<AnswerMultiChoiceWidget> createState() =>
+      _AnswerMultiChoiceWidgetState();
+}
+
+class _AnswerMultiChoiceWidgetState extends State<AnswerMultiChoiceWidget> {
+  final List<String> _answers = [];
+  final List<int> _checkedIndex = [];
+
+  void _onCheckboxChecked(
+    bool? isChecked,
+    int index,
+    String text,
+  ) {
+    setState(() {
+      if (isChecked == true) {
+        _answers.add(widget.question.answers[index].text ?? "");
+        _checkedIndex.add(index);
+      } else {
+        _answers.remove(widget.question.answers[index].text ?? "");
+        _checkedIndex.remove(index);
+      }
+    });
+    widget.onChecked(_answers);
```

Comment:
And we can create the selected answers before invoking the Function 😇 
```suggestion
    final answers = widget.question.answers
        .where((answer) => _checkedIds.contains(answer.id))
        .map((answer) => answer.text ?? "")
        .toList();

    widget.onChecked(answers);
```

Reply:
@ryan-conway By this way, it will make a loop to where and map every time the user makes some change at the checkbox. 🤔  Do you think it affects the performance? 🤔

Reply:
Hmm good point 🤔 What if we just returned the set of ID's in the callback function, and let the viewmodel compute it when we submit the survey? 💡

Reply:
@ryan-conway Yup I think so. 👍🏻